### PR TITLE
[Refactor] Discard unknown transaction status

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

auto discard unknown transaction status can facilitate us to rollback when we support other transaction(PRECOMMIT) status in the future

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 4 (25.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionStatus.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|1|3|33.33%|

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 1]
```diff
@@ -582,6 +582,10 @@ public void readFields(DataInput in) throws IOException {
                 LOG.info("discard expired transaction state: {}", transactionState);
                 continue;
             }
+            if (transactionState.getTransactionStatus() == TransactionStatus.UNKNOWN) {
```
[the remark this answers is not in the document]
Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:585]: > Only one break or one continue statement is acceptable in a loop
seems else if can not make it happy

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 4 (25.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionStatus.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|1|3|33.33%|

**Comment by padmejin** [comment]
You should modify `com.starrocks.transaction.DatabaseTransactionMgr#replayUpsertTransactionState()`.

**Review by chaoyli — DISMISSED** [review]

**Comment by gengjun-git** [comment]
> You should modify `com.starrocks.transaction.DatabaseTransactionMgr#replayUpsertTransactionState()`.

It seems there is no need to modify this function for compatibility. Can you show the reason?

**Comment by padmejin** [comment]
> > You should modify `com.starrocks.transaction.DatabaseTransactionMgr#replayUpsertTransactionState()`.
> 
> It seems there is no need to modify this function for compatibility. Can you show the reason?

I assumed that if there were txn with state unknown in the checkpoint, it must be in journal log as well.

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 2]
```diff
@@ -581,6 +581,9 @@ public void readFields(DataInput in) throws IOException {
             if (transactionState.isExpired(now)) {
                 LOG.info("discard expired transaction state: {}", transactionState);
                 continue;
+            } else if (transactionState.getTransactionStatus() == TransactionStatus.UNKNOWN) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:584]: You should also ignore the UNKNOWN STATUS txn in the DatabaseTransactionMgr.unprotectUpsertTransactionState

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 8 (25.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionStatus.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|4|25.00%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|1|3|33.33%|

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 3]
```diff
@@ -1273,6 +1273,11 @@ public void abortTimeoutTxns(long currentMillis) {
     public void replayUpsertTransactionState(TransactionState transactionState) {
         writeLock();
         try {
+            if (transactionState.getTransactionStatus() == TransactionStatus.UNKNOWN) {
+                LOG.info("remove unknown transaction: {}", transactionState);
+                deleteTransaction(transactionState);
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1278]: this function deleteTransaction() only delete the final state txn, maybe you should write another function to delete the txn from the running map

Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1278]: Why does it appear in the running map?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 7 (28.57%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionStatus.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|1|3|33.33%|

**Review by gengjun-git — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**